platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/SyncGossip.java

```diff
@@ -337,38 +344,24 @@ private void buildSyncProtocolThreads(
                 () -> intakeQueueSizeSupplier.getAsLong() >= eventConfig.eventIntakeQueueThrottleSize(),
                 Duration.ZERO,
                 syncMetrics,
-                platformStatusManager);
+                currentPlatformStatus::get);
         final ProtocolFactory reconnectProtocolFactory = new ReconnectProtocolFactory(
                 platformContext,
                 threadManager,
                 reconnectThrottle,
-                () -> latestCompleteState.getState("SwirldsPlatform: ReconnectProtocol"),
+                getLatestCompleteState,
                 reconnectConfig.asyncStreamTimeout(),
                 reconnectMetrics,
                 reconnectController,
                 new DefaultSignedStateValidator(platformContext),
                 fallenBehindManager,
-                platformStatusManager,
-                platformContext.getConfiguration());
-        final ProtocolFactory emergencyReconnectProtocolFactory = new EmergencyReconnectProtocolFactory(
-                platformContext,
-                threadManager,
-                notificationEngine,
-                emergencyRecoveryManager,
-                reconnectThrottle,
-                emergencyStateSupplier,
-                reconnectConfig.asyncStreamTimeout(),
-                reconnectMetrics,
-                reconnectController,
-                platformStatusManager,
+                currentPlatformStatus::get,
                 platformContext.getConfiguration());
         final ProtocolFactory heartbeatProtocolFactory = new HeartbeatProtocolFactory(
                 Duration.ofMillis(syncConfig.syncProtocolHeartbeatPeriod()), networkMetrics, platformContext.getTime());
-        final HashCompareHandshake hashCompareHandshake =
-                new HashCompareHandshake(epochHash, !protocolConfig.tolerateMismatchedEpochHash());
         final VersionCompareHandshake versionCompareHandshake =
-                new VersionCompareHandshake(appVersion, !protocolConfig.tolerateMismatchedEpochHash());
-        final List<ProtocolRunnable> handshakeProtocols = List.of(hashCompareHandshake, versionCompareHandshake);
+                new VersionCompareHandshake(appVersion, !protocolConfig.tolerateMismatchedVersion());
```

Comment:
Is this really a bug? I'm just removing emergency reconnect stuff.

Reply:
```
final VersionCompareHandshake versionCompareHandshake =
                new VersionCompareHandshake(appVersion, !protocolConfig.tolerateMismatchedEpochHash());
```

`VersionCompareHandshake` is being constructed with the `tolerateMismatchedEpochHash` config value, instead of `tolerateMismatchedVersion`. Bug introduced in https://github.com/hashgraph/hedera-services/pull/11920

Reply:
issue resolved offline